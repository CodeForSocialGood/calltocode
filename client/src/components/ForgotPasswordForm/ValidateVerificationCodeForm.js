import React from 'react'
import {Field, reduxForm} from 'redux-form'
import styles from './ForgotPasswordForm.scss'
import PropTypes from 'prop-types'

function CodeField ({ input }) {
  return (
    <input
      className={styles.inputEmail}
      placeholder="Code"
      {...input} />
  )
}

const ValidateVerificationCodeForm = props => {
  const {handleSubmit} = props
  return (

    <form onSubmit={handleSubmit}
      className={styles.form}>

      <h1
        className={styles.title}>
      Forgot Password?
      </h1>

      <h3>{"Let's validate your code!"}</h3>

      <Field
        name="code"
        component={CodeField} />

      <button
        className={styles.buttonSubmit}
        type="submit">
      Validate code
      </button>
    </form>
  )
}

CodeField.propTypes = {
  input: PropTypes.object
}
ValidateVerificationCodeForm.propTypes = {
  handleSubmit: PropTypes.func
}

export default reduxForm({
  form: 'ForgotPasswordForm', // <------ same form name
  destroyOnUnmount: false, // <------ preserve form data
  forceUnregisterOnUnmount: true // <------ unregister fields on unmount
})(ValidateVerificationCodeForm)
