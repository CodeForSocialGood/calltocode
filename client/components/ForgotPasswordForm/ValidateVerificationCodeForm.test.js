/* eslint-env jest */
import React from 'react'
import {shallow} from 'enzyme'

import ValidateVerificationCodeForm from './ValidateVerificationCodeForm'
let mockProps

beforeEach(() => {
  mockProps = {
    code: null,
    onChangeCode: () => {},
    handleSubmit: () => {}
  }
})

describe('<ValidateVerificationCodeForm />', () => {
  describe('with props.code length 0', () => {
    beforeEach(() => { mockProps.code = [] })

    test('component renders correctly', () => {
      const component = shallow(<ValidateVerificationCodeForm {...mockProps} />)
      expect(component).toMatchSnapshot()
    })
  })

  describe('with props.code length not 0', () => {
    beforeEach(() => { mockProps.code = ['some code'] })

    test('component renders correctly', () => {
      const component = shallow(<ValidateVerificationCodeForm {...mockProps} />)
      expect(component).toMatchSnapshot()
    })
  })
})
