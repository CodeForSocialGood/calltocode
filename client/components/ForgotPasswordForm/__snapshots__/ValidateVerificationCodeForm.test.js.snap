// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ValidateVerificationCodeForm /> with props.code length 0 component renders correctly 1`] = `
<Connect(Form(ValidateVerificationCodeForm))
  code={Array []}
  destroyOnUnmount={false}
  enableReinitialize={false}
  forceUnregisterOnUnmount={true}
  form="ForgotPasswordForm"
  getFormState={[Function]}
  handleSubmit={[Function]}
  keepDirtyOnReinitialize={false}
  onChangeCode={[Function]}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
/>
`;

exports[`<ValidateVerificationCodeForm /> with props.code length not 0 component renders correctly 1`] = `
<Connect(Form(ValidateVerificationCodeForm))
  code={
    Array [
      "some code",
    ]
  }
  destroyOnUnmount={false}
  enableReinitialize={false}
  forceUnregisterOnUnmount={true}
  form="ForgotPasswordForm"
  getFormState={[Function]}
  handleSubmit={[Function]}
  keepDirtyOnReinitialize={false}
  onChangeCode={[Function]}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
/>
`;
